lib/templates.js

```diff
@@ -31,12 +34,16 @@ function renderLayer (data = {}, { params = {}, query = {} } = {}) {
   const json = _.cloneDeep(templates.layer)
   const metadata = data.metadata || {}
   const capabilities = data.capabilities || {}
+  // force an empty array of features if missing
+  if (!data.features) {
+    data.features = []
+  }
```

Comment:
I'm worried about side-effects of mutating the input `data` argument (not convinced we capture everything with our tests). Is this really needed? It looks like `getGeometryTypeFromGeojson` already handles the case where `features` is not defined.

Reply:
This needs to be done or the extent issue as to why it throws an exception needs to be looked into. As it was existing code, it seemed like a TODO vs something to fix for this.

Reply:
Can you link me to where

```js
if (!data.features) {
    data.features = []
}
```
is found in the existing code?